tests/FSharpLint.Core.Tests/Rules/TestTypographyRules.fs

```diff
@@ -207,4 +217,24 @@ type TestTypography() =
     member this.NewLineOnEndOfFile() =
         this.Parse ("let dog = 9" + System.Environment.NewLine)
 
-        Assert.IsTrue(this.ErrorExistsAt(2, 0))
\ No newline at end of file
+        Assert.IsTrue(this.ErrorExistsAt(2, 0))
+
+    [<Test>]
+    member this.``Error for incorrect indentation``() =
+        this.Parse """
+module P
+
+let x =
+  x"""
+
+        Assert.IsTrue(this.ErrorExistsAt(5, 0))
+
+    [<Test>]
+    member this.``No error for correct indentation``() =
```

Comment:
Would also be worth adding a test case for places where the lines may not necessarily be indented as per the rest of the file e.g. below on line 3:
```f#
let rainbow =
    { Boss = "Jeffrey"
      Lackeys = ["Zippy"; "George"; "Bungle"] }
```

Reply:
Yes this should be added. In fact, my current implementation will fail for this case. I'll add a test for this and try to fix it.

Do you know of anything else which would lead to different indentation like this?

Reply:
I've updated the implementation and added some tests for this case.

Reply:
Thanks for the addition of the tests, there's some good docs on indentation here: https://github.com/fsprojects/fantomas/blob/master/docs/FormattingConventions.md#general-rules-for-indentation these could be good test cases. Unfortunately looking at that there's a bunch of additional edge cases :(

Reply:
Thanks for the link Matt, very useful. I looked through it and added handling and test cases for all the exceptional indentation that I saw. Specifically:
- Record field indentation
- Array item indentation
- Pipelining
- Object expressions

There is also the exceptional case of aligning multiple usages of an infix operator, but I'm not sure how to implement this. Did I miss any other cases?

Reply:
No problem. The changes look good, I can't think of any other cases off hand, I think it will be difficult to cover all cases up front, I'll enable it for my projects and see if I spot anything :)